src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -0,0 +1,95 @@
+{
+    CMutableTransaction tx;
+    tx.vout.resize(nInput + 1);
+    tx.vout[nInput].nValue = nValue;
+    tx.nLockTime = lockTime; // all transactions get different hashes
+    CInputCoin coin(MakeTransactionRef(tx), nInput);
+    coin.m_input_bytes = nInputBytes;
+    input_coins.push_back(coin);
+}
+
+// Randomly distribute coins to OutputGroups
+static void GroupCoins(FuzzedDataProvider& fuzzed_data_provider, std::vector<OutputGroup>& output_groups, const std::vector<CInputCoin>& coins, const CoinSelectionParams& coin_params, const bool positive_only)
+{
+    auto output_group = OutputGroup(coin_params);
+    bool valid_outputgroup{false};
+    for (auto& coin : coins) {
+        output_group.Insert(coin, /*depth=*/0, /*from_me=*/true, /*ancestors=*/0, /*descendants=*/0, positive_only);
+        valid_outputgroup = !positive_only || output_group.GetSelectionAmount() > 0;
+        if (valid_outputgroup && fuzzed_data_provider.ConsumeBool()) {
+            output_groups.push_back(output_group);
+            output_group = OutputGroup(coin_params);
+            valid_outputgroup = false;
+        }
+    }
+    if (valid_outputgroup) output_groups.push_back(output_group);
+    return;
+}
+
+FUZZ_TARGET(coinselection)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    std::vector<CInputCoin> utxo_pool;
+
+    const CFeeRate long_term_fee_rate{CFeeRate(ConsumeMoney(fuzzed_data_provider, /*max=*/COIN))};
+    const CFeeRate effective_fee_rate{CFeeRate(ConsumeMoney(fuzzed_data_provider, /*max=*/COIN))};
+    const CAmount cost_of_change{ConsumeMoney(fuzzed_data_provider, /*max=*/COIN)};
+    const CAmount target{fuzzed_data_provider.ConsumeIntegralInRange<CAmount>(1, MAX_MONEY)};
+    const bool subtract_fee_outputs{fuzzed_data_provider.ConsumeBool()};
+
+    auto coin_params = CoinSelectionParams();
+    coin_params.m_subtract_fee_outputs = subtract_fee_outputs;
+    coin_params.m_long_term_feerate = long_term_fee_rate;
+    coin_params.m_effective_feerate = effective_fee_rate;
+
+    // Create some coins
+    CAmount total_balance{0};
+    int nextLockTime{0};
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 10000)
+    {
+        const int nInput{fuzzed_data_provider.ConsumeIntegralInRange<int>(0, 10)};
+        const int nInputBytes{fuzzed_data_provider.ConsumeIntegralInRange<int>(100, 10000)};
+        const CAmount amount{fuzzed_data_provider.ConsumeIntegralInRange<CAmount>(1, MAX_MONEY)};
+        AddCoin(utxo_pool, amount, nInput, nInputBytes, ++nextLockTime);
+        total_balance += amount;
+    }
+
+    std::vector<OutputGroup> group_pos;
+    GroupCoins(fuzzed_data_provider, group_pos, utxo_pool, coin_params, /*positive_only=*/true);
+    std::vector<OutputGroup> group_all;
+    GroupCoins(fuzzed_data_provider, group_all, utxo_pool, coin_params, /*positive_only=*/false);
+
+    // Run coinselection algorithms
+    const auto result_bnb = SelectCoinsBnB(group_pos, target, cost_of_change);
```

Comment:
I think the waste should be computed for this result as well.

Reply:
The call was moved inside of `SelectCoinsBnB()` with #24530 so it is being computed already and wouldn't add coverage - or do you mean another explicit call from the fuzz test?

Reply:
oh yeah, forgot about that.